modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2121,6 +2191,11 @@ public CodegenModel fromModel(String name, Schema schema) {
             return null;
         }
 
+        if (ModelUtils.isAnyTypeSchema(schema)) {
+            // "Any type" means the payload can be any type, e.g. integer, number, object, array...
+            return getAnyTypeModel(name, schema);
+        }
```

Comment:
What if we have AnyTypeSchema in:
- a Model Field
- a Model
- Inline in a response definition?
How about instead:
- pre-processing the the openapi document to check for the locations that need to be fixed
- then insert qty 1 anyTypeModel:
```
AnyType:
  anyOf:
  - type: "null"
  - type: integer
  - type: array ...
```
- link all references that require it to that $ref
Then we don't need any super specific logic like you are adding in python-experimental.